core/src/main/java/org/elasticsearch/index/query/CommonTermsQueryBuilder.java

```diff
@@ -418,7 +419,7 @@ static Query parseQueryString(ExtendedCommonTermsQuery query, Object queryString
         }
 
         if (count == 0) {
-            return null;
+            return new MatchNoDocsQuery("Zero tokens in analyzed query.");
         }
```

Comment:
Probably, I haven't checked what then happens to the clause-less ExtendedCommonTermsQuery. Do you know what happens? I can try that in some tests but let me know if you already know if that would also work.